pkg/validate/validate_component_test.go

```diff
@@ -15,405 +15,249 @@
 package validate
```

Comment:
As a general note, you're doing a bunch of refactoring while also fixing a bug. It's much better to do these in separate PRs. most of my issues are with your test refactoring.